### ISSUE TITLE
Error when creating a c-callable library on MacOS 

### ISSUE THREAD

**Issue by roman-schaerer** [JuliaLang/PackageCompiler.jl#932]
I'm using Julia version 1.10.2 on MacOS Sonoma 14.4. When running PackageCompiler.create_library (based on the example code https://github.com/JuliaLang/PackageCompiler.jl/tree/master/examples/MyLib) as

```julia
PackageCompiler.create_library(".", target_dir;
                                lib_name="my_lib",
                                precompile_execution_file=["$(@__DIR__)/generate_precompile.jl"],
                                precompile_statements_file=["$(@__DIR__)/additional_precompile.jl"],
                                incremental=false,
                                filter_stdlibs=false,
                                force=true, # Overwrite target_dir.
                                header_files = ["$(@__DIR__)/my_lib.h"],
                            )
```
the process errors with 
> ERROR: System image file failed consistency check: maybe opened the wrong version?

The same error occurs for clang 15.0, as well as gcc 13.2.0.

Maybe this issue is related to: [https://github.com/JuliaLang/PackageCompiler.jl/issues/738](https://github.com/JuliaLang/PackageCompiler.jl/issues/738)? There a workaround has been suggested, however I don't know if it is applicable to the create_library function.

**Comment by sloede** [comment]
Yes, this is essentially the same issue (which I also encountered). I am looking forward to #930 hopefully fixing it for both.

**Comment by DilumAluthge** [comment]
This looks like a duplicate of #738, which should be fixed in the recently-released PackageCompiler v2.1.19.

@roman-schaerer Can you try out PackageCompiler v2.1.19 and let me know if it fixes the bug for you?

If your issue still occurs on PackageCompiler v2.1.19, let me know, and I will re-open this issue.

**Comment by DilumAluthge** [comment]
Duplicate of #738
